src/scanner/images/index.ts

```diff
@@ -41,15 +41,15 @@ export async function removePulledImages(images: IPullableImage[]): Promise<void
 }
 
 // Exported for testing
-export function getImageParts(imageWithTag: string) : {imageName: string, imageTag: string} {
+export function getImageParts(imageWithTag: string) : {imageName: string, imageTag: string, imageDigest: string} {
   // we're matching pattern: <registry:port_number>(optional)/<image_name>(mandatory):<image_tag>(optional)@<tag_identifier>(optional)
   // extracted from https://github.com/docker/distribution/blob/master/reference/regexp.go
   const regex = /^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?\/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:\/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][A-Fa-f0-9]{32,}))?$/ig;
   const groups  = regex.exec(imageWithTag);
-  
+
   if(!groups){
     logger.error({image: imageWithTag}, 'Image with tag is malformed, cannot extract valid parts');
-    return {imageName: imageWithTag, imageTag: ''};
+    return { imageName: imageWithTag, imageTag: '', imageDigest: '' };
```

Comment:
This seems like a bit of an error if we returned an empty `imageDigest` considering the rest of the code flow is going to depend on it! Is it possible to fall back to `imageName`/`imageWithTag` when `imageDigest` is empty? Otherwise we might show users empty results or something further up the stack is going to throw when it cannot parse the digest.

Reply:
This verifies this kind of cases, when name is malformed

https://github.com/snyk/kubernetes-monitor/blob/40c8ff4f1f2aabbb0eef861862ef59d6105b4cb4/test/unit/scanner/images.test.ts#L84

Reply:
Correct, but it also returns an empty `imageDigest` when it's malformed. So what will happen if we call `skopeoCopy(emptyImageDigest)`? Previously we would try to pull by `imageName` which was always present and not empty

Reply:
You mean like `skopeoCopy("nginx")` e.g.?

Reply:
Plus, this flow happens way after `skopeoCopy` functionality.